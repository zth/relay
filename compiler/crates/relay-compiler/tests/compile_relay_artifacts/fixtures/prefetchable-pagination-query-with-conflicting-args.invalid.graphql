# expected-to-throw

query prefetchablePaginationQueryWithConflictingArgsQuery($site: String) {
  node(id: "x") {
    ...prefetchablePaginationQueryWithConflictingArgs_FirstFragment @arguments(site: $site)
  }
}

fragment prefetchablePaginationQueryWithConflictingArgs_FirstFragment on Node
  @refetchable(queryName: "RefetchableFragmentQuery")
  @argumentDefinitions(
    count: {type: "Int", defaultValue: 10}
    cursor: {type: "ID"}
    site: {type: "String"}
  ) {
  id
  ...prefetchablePaginationQueryWithConflictingArgs_SecondFragment
  ... on User {
    name
    friends(after: $cursor, first: $count)
    @connection(key: "PaginationFragment_friends", prefetchable_pagination: true) {
    edges {
      node {
        id
        # local $site
        url(site: $site)
      }
    }
  }
  }
}

fragment prefetchablePaginationQueryWithConflictingArgs_SecondFragment on Node {
  # global $site
  p2: url(site: $site)
}
