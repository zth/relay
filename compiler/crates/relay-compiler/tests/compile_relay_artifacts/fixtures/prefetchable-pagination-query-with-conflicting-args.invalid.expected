==================================== INPUT ====================================
# expected-to-throw

query prefetchablePaginationQueryWithConflictingArgsQuery($site: String) {
  node(id: "x") {
    ...prefetchablePaginationQueryWithConflictingArgs_FirstFragment @arguments(site: $site)
  }
}

fragment prefetchablePaginationQueryWithConflictingArgs_FirstFragment on Node
  @refetchable(queryName: "RefetchableFragmentQuery")
  @argumentDefinitions(
    count: {type: "Int", defaultValue: 10}
    cursor: {type: "ID"}
    site: {type: "String"}
  ) {
  id
  ...prefetchablePaginationQueryWithConflictingArgs_SecondFragment
  ... on User {
    name
    friends(after: $cursor, first: $count)
    @connection(key: "PaginationFragment_friends", prefetchable_pagination: true) {
    edges {
      node {
        id
        # local $site
        url(site: $site)
      }
    }
  }
  }
}

fragment prefetchablePaginationQueryWithConflictingArgs_SecondFragment on Node {
  # global $site
  p2: url(site: $site)
}
==================================== ERROR ====================================
✖︎ Fragment variable `$site` conflicts with a global variable generated by the @refetchable generated query

  prefetchable-pagination-query-with-conflicting-args.invalid.graphql:14:5
   13 │     cursor: {type: "ID"}
   14 │     site: {type: "String"}
      │     ^^^^
   15 │   ) {
